src/main/java/sirius/web/http/Response.java

```diff
@@ -812,11 +799,17 @@ public void redirectToGet(String url) {
      * @param url the URL to redirect to
      */
     public void redirectPermanently(String url) {
+        performRedirect(url, HttpResponseStatus.MOVED_PERMANENTLY);
+    }
+
+    private void performRedirect(String url, HttpResponseStatus status) {
         if (cacheSeconds == null || cacheSeconds == 0) {
             userMessagesCache.cacheUserMessages(webContext);
         }
 
-        HttpResponse response = createFullResponse(HttpResponseStatus.MOVED_PERMANENTLY, true, Unpooled.EMPTY_BUFFER);
+        setDateAndCacheHeaders(System.currentTimeMillis(), cacheSeconds == null ? 0 : cacheSeconds, isPrivate);
```

Comment:
shouldn't we just leave this out for 0/null - otherwise this would be a semantic change to the current code and would require intense testing...